pagefaultgames/pokerogue
# [Bug] Fix run away ability

## What are the changes?
Pokemon with the ability Run Away are not able to flee from battles where an opposing pokemon has a trapping ability.

## Why am I doing these changes?
A Pokémon with the Run Away ability can always flee and use Teleport successfully, regardless of trapping moves and abilities.
- https://bulbapedia.bulbagarden.net/wiki/Run_Away_(Ability)#Effect

## What did change?
`ArenaTrapAbAttr` in `ability.ts` was modified to add a check for the ability `Run_Away`.

### Screenshots/Videos
The attached video shows a pokemon with the ability Run Away successfully running from pokemon with the ability Arena Trap and the passive Shadow Tag.

https://github.com/pagefaultgames/pokerogue/assets/167576411/c8c3981d-3c33-4cf1-a696-f2e66d8d9af9

## How to test the changes?
Start a new game with a pokemon with the ability Run Away.
Set the following overrides in `overrides.ts` :
- `OPP_ABILITY_OVERRIDE` to `Abilities.ARENA_TRAP` or `Abilities.SHADOW_TAG`

Using each pokemon attempt to run from battles.
You should successfully escape everytime, unless you are in a trainer battle in Classic or an x50 battle in Endless.

## Checklist
